table/block_based_table_reader.cc

```diff
@@ -1436,23 +1504,20 @@ BlockBasedTable::BlockEntryIteratorState::NewSecondaryIterator(
   BlockHandle handle;
   Slice input = index_value;
   Status s = handle.DecodeFrom(&input);
-  auto iter = NewDataBlockIterator(table_->rep_, read_options_, handle, nullptr,
-                                   is_index_, s);
-  if (block_cache_cleaner_) {
-    uint64_t offset = handle.offset();
-    {
-      ReadLock rl(&cleaner_mu);
-      if (cleaner_set.find(offset) != cleaner_set.end()) {
-        // already have a reference to the block cache objects
-        return iter;
-      }
+  auto rep = table_->rep_;
+  if (block_map_) {
+    auto block = block_map_->find(handle.offset());
+    assert(block != block_map_->end());
+    if (LIKELY(block != block_map_->end())) {
```

Comment:
But a bug here would only cause performance issues and does not qualify taking down the service for it. I think we should understand that at each point of time there are many bugs in our code base and we should provide mechanisms to catch them but we do not have to crash the system each time one shows up.
About the existing approach here, I believe we have monitoring that aggregates the errors reflected on logs. Whether we use that often enough is one thing but using logs to record some unexpected behaviors can still give visibility for non-fatal errors.